tests/test_optimizer.py

```diff
@@ -66,3 +69,116 @@ def test_grad_averager():
     # after no longer use_averaged_gradients
     assert not torch.allclose(model1.w.grad, ref_average)
     assert not torch.allclose(model2.w.grad, ref_average)
+
+
+@pytest.mark.forked
+@pytest.mark.parametrize("offload_optimizer, reuse_tensors", [(False, False), (True, False), (False, True)])
+def test_state_averager(offload_optimizer, reuse_tensors):
+    dht1 = hivemind.DHT(start=True)
+    dht2 = hivemind.DHT(initial_peers=dht1.get_visible_maddrs(), start=True)
+
+    torch.manual_seed(1337)
+    torch.use_deterministic_algorithms(True)
+    model1 = nn.Linear(2, 3)
+    model2 = nn.Linear(2, 3)
+
+    extras1 = (torch.randn(2, 2), -torch.rand(1))
+    extras2 = (-torch.randn(2, 2), torch.rand(1))
+
+    avgr1 = TrainingStateAverager(
+        dht=dht1,
+        param_groups=model1.parameters(),
+        optimizer=partial(torch.optim.Adam, lr=0.1, betas=(0.9, 0.9)),
+        scheduler=partial(torch.optim.lr_scheduler.LambdaLR, lr_lambda=lambda t: 1.0 / max(1, t)),
+        average_opt_statistics=("exp_avg_sq",),
+        offload_optimizer=offload_optimizer,
+        reuse_tensors=reuse_tensors,
+        extra_tensors=extras1,
+        target_group_size=2,
+        prefix="my_exp",
+        start=True,
+    )
+
+    avgr2 = TrainingStateAverager(
+        dht=dht2,
+        param_groups=model2.parameters(),
+        optimizer=partial(torch.optim.Adam, lr=0.1, betas=(0.9, 0.9)),
+        scheduler=partial(torch.optim.lr_scheduler.LambdaLR, lr_lambda=lambda t: 1.0 / max(1, t)),
+        average_opt_statistics=("exp_avg_sq",),
+        offload_optimizer=offload_optimizer,
+        reuse_tensors=reuse_tensors,
+        extra_tensors=extras2,
+        target_group_size=2,
+        prefix="my_exp",
+        start=True,
```

Comment:
nit: Let's move `optimizer`, `scheduler`, etc. to kwargs and pass `**kwargs` to both averagers. It would be easier to see that they are the same + we'd have less code.

Reply:
fixd, thanks!